localstack/services/s3/s3_listener.py

```diff
@@ -1585,5 +1519,42 @@ def authenticate_presign_url_signv4(method, path, headers, data, url, query_para
         )
 
 
+def serve_static_website(headers, path, bucket_name):
+    s3_client = aws_stack.connect_to_service('s3')
+
+    # check if bucket exists
+    try:
+        s3_client.head_bucket(Bucket=bucket_name)
+    except ClientError:
+        return no_such_bucket(bucket_name, headers.get('x-amz-request-id'), 404)
+
+    try:
+        if path != '/':
+            res = s3_client.head_object(Bucket=bucket_name, Key=path)
+            if res['ResponseMetadata']['HTTPStatusCode'] == 200:
+                content = s3_client.get_object(Bucket=bucket_name, Key=path)['Body'].read()
+                return requests_response(status_code=200, content=content)
+    except ClientError:
+        LOGGER.debug('No such key found. %s' % path)
+
+    website_config = s3_client.get_bucket_website(Bucket=bucket_name)
+    path_suffix = website_config.get('IndexDocument', {}).get('Suffix', '').lstrip('/')
+    index_document = '%s/%s' % (path.rstrip('/'), path_suffix)
+    try:
+        res = s3_client.head_object(Bucket=bucket_name, Key=index_document)
```

Comment:
nitpick: Instead of doing `head_object` and then `get_object`, we could simply go directly with `get_object`, and catch any errors if the object does not exist (for slightly improved performance). Same on lines 1533 and 1551.